rust_src/src/frames.rs

```diff
@@ -225,14 +227,19 @@ pub fn framep(object: LispObject) -> LispObject {
 /// use `display-graphic-p' or any of the other `display-*-p'
 /// predicates which report frame's specific UI-related capabilities.
 #[lisp_fn(min = "0")]
-pub fn window_system(frame: Option<LispFrameRef>) -> Option<LispSymbolRef> {
-    let f = frame.map_or(selected_frame(), |f| f.as_lisp_obj());
-    let window_system = framep(f);
+pub fn window_system(frame: Option<LispFrameRef>) -> LispObject {
+    let frame = frame.unwrap_or_else(|| selected_frame().as_frame_or_error());
```

Comment:
Do we really need to check if selected_frame is a frame ? If we do I think an assertion would be better.